yt/visualization/plot_window.py

```diff
@@ -2014,3 +2016,75 @@ def SlicePlot(ds, normal=None, fields=None, axis=None, *args, **kwargs):
             del kwargs['north_vector']
 
         return AxisAlignedSlicePlot(ds, normal, fields, *args, **kwargs)
+
+class LinePlot(PlotMPL):
+    r"""
+    A class for constructing line plots
+
+    Parameters
+    ----------
+
+    ds : :class:`yt.data_objects.static_output.Dataset`
+        This is the dataset object corresponding to the
+        simulation output to be plotted.
+    fields : string
+        The name(s) of the field(s) to be plotted.
+    point1: tuple
+        Contains the coordinates of the first point for constructing the line
+    point2: tuple
+        Contains the coordinates of the second point for constructing the line
+    resolution: int
+        How many points to sample between point1 and point2 for constructing
+        the line plot
+    """
+
+    def __init__(self, ds, fields, point1, point2, resolution,
+                 figure_size=5., aspect=None, fontsize=18.):
+        handler = CartesianCoordinateHandler(ds)
```

Comment:
You probably want to use `ds.coordinates` here instead of creating a new `CartesianCoordinateHandler`. In case you didn't know:

```
In [1]: type(ds.coordinates)
Out[1]: yt.geometry.coordinates.cartesian_coordinates.CartesianCoordinateHandler
```

Also keep in mind that curvilinear coordinates are a thing as well.